kademlia/src/main/java/ingsw/group1/kademlia/pendingrequests/PendingRequestManager.java

```diff
@@ -0,0 +1,259 @@
+import ingsw.group1.kademlia.listeners.StoreResultListener;
+import ingsw.group1.msglibrary.SMSPeer;
+import ingsw.group1.repnetwork.StringResource;
+
+/**
+ * Class meant to handle running operations in the Network.
+ * Does not answer to incoming requests, only handles response to sent requests.
+ * A maximum number of Requests can be set during creation.
+ *
+ * @author Riccardo De Zen
+ * CODE REVIEW
+ */
+public class PendingRequestManager {
+
+    private static final int MAX_SIZE = KadAction.MAX_ID;
+    private static final int SENTINEL_ID = -1;
+    private static final String CONTINUE_ERR = "Provided Action is not a Response.";
+    private static final String SIZE_ERR = "The capacity should not be higher than: " + MAX_SIZE;
+
+    private final int capacity;
+
+    private ActionPropagator defaultActionPropagator;
+    private NodeDataProvider<BinarySet, PeerNode> defaultNodeDataProvider;
+    private List<PendingRequest> currentRequests = new ArrayList<>();
+
+    /**
+     * Constructor.
+     *
+     * @param capacity                The maximum amount of PendingRequests that can be pending
+     *                                at a time. Can't
+     *                                be higher than {@link PendingRequestManager#MAX_SIZE}.
+     * @param defaultActionPropagator The {@link ActionPropagator} that will be used by default
+     *                                by enqueued {@code PendingRequests}.
+     * @param defaultNodeDataProvider The {@link NodeDataProvider} that will be used by default
+     *                                by enqueued {@code PendingRequests}.
+     */
+    public PendingRequestManager(
+            int capacity,
+            @NonNull ActionPropagator defaultActionPropagator,
+            @NonNull NodeDataProvider<BinarySet, PeerNode> defaultNodeDataProvider
+    ) {
+        if(capacity > MAX_SIZE)
+            throw new IllegalArgumentException(SIZE_ERR);
+        this.capacity = capacity;
+        this.defaultActionPropagator = defaultActionPropagator;
+        this.defaultNodeDataProvider = defaultNodeDataProvider;
+    }
+
+    /**
+     * Setter for {@link PendingRequestManager#defaultActionPropagator}.
+     *
+     * @param newDefaultPropagator the new value for the default propagator.
+     */
+    public void setDefaultActionPropagator(@NonNull ActionPropagator newDefaultPropagator) {
+        defaultActionPropagator = newDefaultPropagator;
+    }
+
+    /**
+     * Setter for {@link PendingRequestManager#defaultNodeDataProvider}.
+     *
+     * @param newDefaultProvider the new value for the default node provider.
+     */
+    public void setDefaultNodeDataProvider(@NonNull NodeDataProvider<BinarySet, PeerNode> newDefaultProvider) {
+        defaultNodeDataProvider = newDefaultProvider;
+    }
+
+    /**
+     * Method to enqueue a {@link PingPendingRequest}.
+     *
+     * @return true if the {@code PendingRequest} could be enqueued, false otherwise.
+     */
```

Comment:
Missing @param statements.
Probably variables' names are self-explicative, but I think it is always better to specify params description.